### ISSUE TITLE
Attempting to combine different files error when using nathantreid:meteor-css-modules

### ISSUE THREAD

**Issue by TimBroddin** [meteor/meteor#6881]
When using Meteor 1.3.2.4 (and all other 1.3.2 releases) and using nathantreid:meteor-css-modules (with SCSS-compilation and server-side rendering) you get the follow error. This issue was not present in 1.3.0.

```
=> Started proxy.                             
=> Started MongoDB.                           
                                              
/Users/timbroddin/.meteor/packages/less/.2.6.0.h53n1v++os+web.browser+web.cordova/plugin.compileLessBatch.os/npm/node_modules/meteor/promise/node_modules/meteor-promise/promise_server.js:116
      throw error;
            ^
Error: Attempting to combine different files:
{ type: 'js',
  sourcePath: 'app/style/style.scss',
  servePath: 'app/style/style.scss',
  installPath: 'app/style/style.scss',
  deps: { './style.css': {} },
  lazy: true,
  data: <Buffer 6d 6f 64 75 6c 65 2e 65 78 70 6f 72 74 73 20 3d 20 72 65 71 75 69 72 65 28 22 2e 2f 73 74 79 6c 65 2e 63 73 73 22 29 3b 0a>,
  hash: '39511a6d70e7980cb52ca5519c549245aa450762' }
{ type: 'js',
  data: <Buffer 65 78 70 6f 72 74 73 2e 5f 5f 65 73 4d 6f 64 75 6c 65 20 3d 20 74 72 75 65 3b 0a 0a 0a 76 61 72 20 73 74 79 6c 65 73 20 3d 20 7b 20 22 6f 75 74 65 72 43 ...>,
  sourcePath: 'app/style/style.scss',
  targetPath: 'app/style/style.scss',
  servePath: '/app/style/style.scss',
  hash: '77e7234e10a2a22276b0fcd12ef09d4c4e98601a',
  sourceMap: undefined,
  lazy: false,
  bare: false,
  mainModule: false,
  imported: true,
  installPath: 'app/style/style.scss' }

    at checkProperty (/tools/isobuild/import-scanner.js:274:15)
    at ImportScanner._combineFiles (/tools/isobuild/import-scanner.js:263:35)
    at /tools/isobuild/import-scanner.js:187:14
    at Array.forEach (native)
    at ImportScanner.addInputFiles (/tools/isobuild/import-scanner.js:158:11)
    at /tools/isobuild/compiler-plugin.js:834:15
    at Array.forEach (native)
    at Function.computeJsOutputFilesMap (/tools/isobuild/compiler-plugin.js:805:19)
    at ClientTarget._emitResources (/tools/isobuild/bundler.js:913:8)
    at /tools/isobuild/bundler.js:684:12
    at /tools/utils/buildmessage.js:359:18
    at [object Object].withValue (/tools/utils/fiber-helpers.js:89:14)
    at /tools/utils/buildmessage.js:352:34
    at [object Object].withValue (/tools/utils/fiber-helpers.js:89:14)
    at /tools/utils/buildmessage.js:350:23
    at [object Object].withValue (/tools/utils/fiber-helpers.js:89:14)
    at Object.enterJob (/tools/utils/buildmessage.js:324:26)
    at ClientTarget.make (/tools/isobuild/bundler.js:675:18)
    at /tools/isobuild/bundler.js:2474:14
    at /tools/isobuild/bundler.js:2563:20
    at Array.forEach (native)
    at Function._.each._.forEach (/Users/timbroddin/.meteor/packages/meteor-tool/.1.3.2_4.hq6may++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/underscore/underscore.js:79:11)
    at /tools/isobuild/bundler.js:2562:7
    at /tools/utils/buildmessage.js:271:13
    at [object Object].withValue (/tools/utils/fiber-helpers.js:89:14)
    at /tools/utils/buildmessage.js:264:29
    at [object Object].withValue (/tools/utils/fiber-helpers.js:89:14)
    at /tools/utils/buildmessage.js:262:18
    at [object Object].withValue (/tools/utils/fiber-helpers.js:89:14)
    at /tools/utils/buildmessage.js:253:23
    at [object Object].withValue (/tools/utils/fiber-helpers.js:89:14)
    at Object.capture (/tools/utils/buildmessage.js:252:19)
    at Object.exports.bundle (/tools/isobuild/bundler.js:2455:31)
    at /tools/runners/run-app.js:591:36
    at Function.run (/tools/tool-env/profile.js:489:12)
    at bundleApp (/tools/runners/run-app.js:581:34)
    at AppRunner._runOnce (/tools/runners/run-app.js:634:35)
    at AppRunner._fiber (/tools/runners/run-app.js:887:28)
    at /tools/runners/run-app.js:411:12
```

A workaround is commenting these lines in `tools/isobuild/compiler_plugin.js`:

```
      // Eager CSS is added unconditionally to a combined <style> tag at
      // the beginning of the <head>. If the corresponding module ever
      // gets imported, its module.exports object should be an empty stub,
      // rather than a <style> node added dynamically to the <head>.
      self.addJavaScript({
        ...options,
        data: "// These styles have already been applied to the document.\n",
        lazy: true
      });
```

The sourcemap generator (or consumer?) seems to choke on them.

I made a repo to demonstrate this issue: https://github.com/TimBroddin/meteor-css-modules-bug (simple import of scss file)

**Comment by benjamn** [comment]
Thanks for the detailed bug report. I see that `nathantreid:meteor-css-modules` is calling both `addStylesheet` and `addJavaScript`, and I believe the code you suggested commenting out is conflicting with his `addJavaScript` call.

The code in `compiler-plugin.js` is only there to make it safe to `import` a stylesheet even if it doesn't do anything (e.g. because it's an eager stylesheet and was already added to the `<head>`). In other words, that empty stub definitely should lose to any legitimate calls to `addJavaScript` by the build plugin that called `addStyleSheet`.

**Comment by nathantreid** [comment]
I was able to work around this for now by specifying a different `sourcePath` when calling `addStylesheet`.

**Comment by benjamn** [comment]
This should be fixed if you do `meteor update --release `1.3.3-beta.0`.
